### PR TITLE
Add support for moving pypy virtualenv, explicitly not supporting pypy3. Test refactoring to come in later review.

### PR DESCRIPTION
**Pull request by joshSzep**
Replacement review for https://github.com/Yelp/virtualenv-tools/pull/11

This extracts just the changes which add the pypy virtualenv movement support and the addition of new tests to cover this functionality. A future review can be posted to refactor the tests to eliminate the huge level of duplication as was done in https://github.com/Yelp/virtualenv-tools/pull/11

### CODE REVIEW

**Comment by joshSzep** [comment]
local tox run https://fluffy.yelpcorp.com/i/cXbSj5BTkzB3nzXLnW123p5NfF3Tc1KC.html

File: tests/virtualenv_tools_test.py [hunk 1]
```diff
@@ -76,28 +76,29 @@ def test_each_part_idempotent(tmpdir, venv, capsys):
...
         'bash', '-c',
-        ". {} && python -c 'import sys; print(sys.executable)'".format(
-            pipes.quote(path.join('bin/activate').strpath),
+        ". {activate} && {py} -c 'import sys; print(sys.executable)'".format(
+            activate=pipes.quote(path.join('bin/activate').strpath),
+            py=py
         )
     )).decode('UTF-8').strip()
-    assert exe == path.join('bin/python').strpath
+    assert exe == path.join('bin/{py}'.format(py=py)).strpath
```
Comment (bukzor) [inline tests/virtualenv_tools_test.py]: the parts where you take pains to change "python" to sometimes-pypy don't feel right
"python" should simply map to pypy in the pypy case
thats >50% of the changes as far as i see

File: tests/virtualenv_tools_test.py [hunk 2]
```diff
@@ -207,3 +208,191 @@ def test_not_a_virtualenv_missing_site_packages(fake_venv, capsys):
...
+# -------- BEGIN Temp PyPy support Tests ------------
+# After base functionality for pypy venv movement is reviewed
+# these can be refactored in a way similar to how it was done in
+# https://github.com/Yelp/virtualenv-tools/pull/11 to reduce duplication
+
+
+skip_test_if_requires_pypy3 = pytest.mark.skipif(
+    sys.version_info >= (3, 3),
+    reason='pypy3 is not supported.'
+)
```
Comment (bukzor) [inline tests/virtualenv_tools_test.py]: You're skipping these tests for cpython3.6 as well.
Doesn't seem right.

What goes wrong if we delete this thing?

- [x] fixed

File: virtualenv_tools.py [hunk 3]
```diff
@@ -178,20 +179,23 @@ def _update_pth_file(pth_filename, orig_path):
         changed = True
         relto_original = os.path.relpath(val, orig_path)
         # lib/pythonX.X/site-packages
-        relto_pth = os.path.join('../../..', relto_original)
+        relto_pth = os.path.join(
+            '..' if is_pypy else '../../..',
+            relto_original
+        )
```
Comment (bukzor) [inline virtualenv_tools.py]: Deserves a comment.
Note that you've separated the `../../..` from its comment.

- [x] fixed

File: virtualenv_tools.py [hunk 4]
```diff
@@ -271,7 +283,51 @@ def _get_original_state(path):
...
+
+    orig_path = get_orig_path(path)
+    return Virtualenv(
+        path=path,
+        bin_dir=bin_dir,
+        lib_dirs=[lib_dir, os.path.join(path, 'lib_pypy')],
+        site_packages=site_packages,
+        orig_path=orig_path,
+        is_pypy=True
+    )
```
Comment (bukzor) [inline virtualenv_tools.py]: This is much too wet. Please DRY.

- [x] fixed

File: virtualenv_tools.py [hunk 5]
```diff
@@ -304,7 +360,9 @@ def main(argv=None):
         return 1
 
     try:
-        venv = _get_original_state(args.path)
+        is_pypy = os.path.isdir(os.path.join(args.path, 'lib_pypy'))
+        venv = _get_original_state_pypy(args.path) \
+            if is_pypy else _get_original_state(args.path)
```
Comment (bukzor) [inline virtualenv_tools.py]: `is_pypy` seems like part of the original state, and as such belongs *inside* the get-original-state function.

Did you know: the cyclomatic complexity of a try-block increases by one for each *operation* it contains.

- [x] fixed

**Review by bukzor — CHANGES_REQUESTED** [review]

File: tests/virtualenv_tools_test.py (hunk 2 above)
Comment (bukzor) [reply tests/virtualenv_tools_test.py]: Clarified with @joshSzep. This allows us to punt on pypy3 support, since we don't have it installed our on dev machines, and actually don't care about it much. Pre-existing policy tests python3 when the test interpreter is python3, by analogy, we'd test pypy3 when test interpreter is python3, but we're punting for now, instead.

File: tests/virtualenv_tools_test.py (hunk 1 above)
Comment (joshSzep) [reply tests/virtualenv_tools_test.py]: Yep this was not necessary to do it this way, fixed

File: tests/virtualenv_tools_test.py (hunk 2 above)
Comment (joshSzep) [reply tests/virtualenv_tools_test.py]: Have improved the explanation around this

File: virtualenv_tools.py (hunk 3 above)
Comment (joshSzep) [reply virtualenv_tools.py]: comments fixed up :)

File: virtualenv_tools.py (hunk 5 above)
Comment (joshSzep) [reply virtualenv_tools.py]: good suggestion, fixed

**Review by bukzor — APPROVED** [review]
Thanks!

File: tests/virtualenv_tools_test.py [hunk 6]
```diff
@@ -100,110 +43,223 @@ def assert_virtualenv_state(path):
...
-    libdir = 'lib/python{}.{}'.format(*sys.version_info[:2])
-    bad_pyc = venv.before.join(libdir, 'bad.pyc')
-    bad_pyc.write_binary(b'I am a very naughty pyc\n')
-    # Retries on failures as well
-    for _ in range(2):
-        with pytest.raises(ValueError):
-            run(venv.before, venv.after)
-        out, _ = capsys.readouterr()
-        assert out == 'Error in {}\n'.format(bad_pyc.strpath)
+class TestVirtualEnvTools(object):
```
Comment (asottile) [inline tests/virtualenv_tools_test.py]: why? this is unrelated to this change (and negative imo)
